File: searchIndex-platform/module/search-api/search-manager/app/controllers/MetricsController.java
package controllers;

import org.ekstep.common.dto.Request;

import managers.PlaySearchManager;
import play.libs.F.Promise;
import play.mvc.Result;

public class MetricsController extends SearchBaseController {

	private PlaySearchManager mgr = new PlaySearchManager();

	public Promise<Result> search() {
		String apiId = "composite-search.metrics";
		Request request = getRequest(request().body(), apiId, request().uri());
		Promise<Result> searchResponse = mgr.metrics(request);
		return searchResponse;
	}

}

File: searchIndex-platform/module/search-api/search-manager/app/controllers/MetricsController.scala
package controllers

import akka.actor.{ActorRef, ActorSystem}
import javax.inject.{Inject, Named}
import play.api.mvc.ControllerComponents

import scala.concurrent.ExecutionContext
import managers.PlaySearchManager

class MetricsController@Inject()(@Named("SearchManager") actor: ActorRef, cc: ControllerComponents, actorSystem: ActorSystem)(implicit exec: ExecutionContext) extends BaseController(cc) {

    private val mgr = new PlaySearchManager

    def search() = Action.async { implicit request =>
        val apiId: String = "composite-search.metrics"
        val internalReq = getRequest(apiId)
        getResult(mgr.metrics(internalReq, actor))

    }
}
